artifacts/definitions/Linux/Events/TrackProcesses.yaml

```diff
@@ -0,0 +1,201 @@
+
+  This event artifact enables the global process tracker and makes it
+  possible to run many other artifacts that depend on the process
+  tracker.
+  
+  auditd is used to track processes, and it must be installed on the
+  system. If InstallAudit is enabled, the artifact will install it
+  using apt-get (which only works on Debian-like operating systems,
+  like Debian and Ubuntu).
+  
+  When the artifact starts, it will insert audit rules that track
+  the syscalls *execve*, *exit* and *exit_group*. These rules are removed
+  when the event artifact is disabled. However, in case they are not
+  removed automatically, for instance if velociraptor were to crash,
+  the following commands (which must be run as root) will remove then:
+  
+  ```
+  auditctl -d exit,always -F arch=b64 -S execve -k procmon
+  auditctl -d exit,always -F arch=b32 -S execve -k procmon
+  auditctl -d exit,always -F arch=b64 -S exit,exit_group -k procmon_exit
+  auditctl -d exit,always -F arch=b32 -S exit,exit_group -k procmon_exit
+  ```
+  
+  Remember to replace the keys if you used keys other than the defaults.
+  
+  Note that processes that are killed or do not shut down properly will
+  note get their exit timestamps registered.
+
+precondition: SELECT OS From info() where OS = 'linux'
+
+type: CLIENT_EVENT
+
+required_permissions:
+  - EXECVE
+
+parameters:
+  - name: InstallAudit
+    type: bool
+    default: False
+    description: Run apt-get update and apt-get install to ensure that auditd is installed
+  - name: AuditKeyExecve
+    default: procmon
+    description: Key to use for execve syscalls. Change to match an existing key if you are already using audit to track processes.
+  - name: AuditKeyExit
+    default: procmon_exit
+    description: Key to use for exit syscalls. Change to match an existing key if you are already using audit to track processes.
+  - name: RemoveRules
+    type: bool
+    default: True
+    description: Remove rules when removing the event artifact
+  - name: AlsoForwardUpdates
+    type: bool
+    description: Upload all tracker state updates to the server
+  - name: MaxSize
+    type: int64
+    description: Maximum size of the in-memory process cache (default 10k)
+  - name: AddEnrichments
+    type: bool
+    description: Calculate hashes on process binaries
+
+sources:
+  - query: |
+     LET _ <= SELECT *
+       FROM execve(argv=['apt-get', '-y', 'update'])
+       WHERE InstallAudit
+     LET _ <= SELECT *
+       FROM execve(
+         argv=['apt-get', '-y', '-o', 'Debug::pkgProblemResolver=yes', '--no-install-recommends', 'auditd'])
+       WHERE InstallAudit
+
+     LET AuditCtl(action, syscalls, key) = SELECT *
+       FROM foreach(
+         row=('b64', 'b32', ),
+         query={
+           SELECT *
+           FROM execve(
+             argv=['auditctl', action, 'exit,always', '-F', 'arch='+_value, '-S', syscalls, '-k', key])
+         })
+
+     LET _ <= SELECT *
```

Comment:
Maybe abstract into a function so we can only call it if auditd is found and installed